library/alloc/src/string.rs

```diff
@@ -1,8 +1,8 @@
-//! A UTF-8 encoded, growable string.
+//! A UTF-8–encoded, growable string.
```

Comment:
I'm not sure about this change; there's no way to confuse this with anything else and the en dash doesn't match the regular dash for UTF-8.

Reply:
I mean, it's not a big deal, but I'm kind of a typographical perfectionist :)

I think the general style is to use an en dash when the first half has a regular dash in it so you can distinguish the two halves. I also used this style for [`std::fmt::Write`](https://doc.rust-lang.org/nightly/std/fmt/trait.Write.html):

![image](https://user-images.githubusercontent.com/37223377/96639001-e38dff00-12d5-11eb-8287-9fde86539352.png)

I do prefer it with the en dash, but I can remove it if you think it doesn't make sense to have here.

Reply:
Well, I think this is why it should be split into a different change. I don't have strong opinions on this but I also don't feel comfortable approving it; it doesn't seem strictly better than before.

Reply:
I'll cherry-pick the lib commit into a separate PR.